### PR TITLE
[Core]Remove checkpoint table

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

This also removes actor_registry from node manager.

**It currently breaks Java**. I tried to remove all Java related parts, but there might be remaining things. Question; @kfstorm @ffbin Can you guys take care of it after we just merge this PR?

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: java/api/src/main/java/io/ray/api/Checkpointable.java [hunk 1]
```diff
@@ -85,16 +85,4 @@ public Checkpoint(UniqueId checkpointId, long timestamp) {
    *     restart from the beginning.
    */
   UniqueId loadCheckpoint(ActorId actorId, List<Checkpoint> availableCheckpoints);
```
Comment (raulchen) [inline java/api/src/main/java/io/ray/api/Checkpointable.java:87]: @chaokunyang this whole class can be removed.

Comment (chaokunyang) [reply java/api/src/main/java/io/ray/api/Checkpointable.java]: removed

**Comment by rkooo567** [comment]
@chaokunyang Now, it seems like everything is failing?

**Comment by chaokunyang** [comment]
@rkooo567 Sorry, I missed a diff when I rebase master to resolve the conflcit for node_manager.h

**Comment by kfstorm** [comment]
> @rkooo567 Sorry, I missed a diff when I rebase master to resolve the conflcit for node_manager.h

Please avoid rebasing when developing a pull request. https://www.atlassian.com/git/tutorials/merging-vs-rebasing#the-golden-rule-of-rebasing

**Comment by rkooo567** [comment]
@chaokunyang Can you see if Java failures are related?

**Comment by chaokunyang** [comment]
> @chaokunyang Can you see if Java failures are related?

Just pushed a fix, I'll check java ci.

**Comment by chaokunyang** [comment]
@rkooo567 All java issues has been resolved

**Comment by rkooo567** [comment]
@raulchen @chaokunyang Can you guys review it now? It passes all tests now.

**Review by chaokunyang — APPROVED** [review]
LGTM

**Review by raulchen — APPROVED** [review]

**Pull request merged**